test/splitter.js

```diff
@@ -162,14 +162,21 @@ contract('Splitter', accounts => {
   });
 
   context('When owner pause the contract', () => {
-    context('if the contract is unpaused', () => {
+    context('if the contract is not paused', () => {
```

Comment:
"not paused" but you pause in `beforeEach`. Not sure what you are trying to communicate.

Reply:
I prefer to put the common logic including `Act` in `beforeEach` because they can be shared in different `it`('Assert'). Because the contract by default is not paused, so we don't need to do `Arrange` here. Please see 79be12b0dccb46a7a75efb963468680ed11c555f

Reply:
I was asking about the wording `'if the contract is not paused'`, which I find does not correlate with what is being done in it.

Reply:
I put `context('if the contract is not paused', () => {` to group the happy case because there is a sad case in the same level `context('if the contract is paused', () => {`. Can you advice how can I improve this?

Reply:
```javascript
context('When owner pause the contract', () => {
    context('if the contract is not paused', () => {
        beforeEach("unpause the contract");
        it("pause the contract");
```
Is this what you want to test?

Reply:
sorry, I can't find `beforeEach("unpause the contract");` in the test.

There is an `Act` in the `beforeEach` like

```
beforeEach(async () => {
        // Arrange, because by default, the contract is not paused, no need to arrange.
        // Act
        tx = await contract.methods.pause().send({from: owner});
      });
```

The test cases I try to cover are
```
context('When owner pause the contract') {
    context('if the contract is not paused') {
        beforeEach() // No need to Arrange, shareable Act for both two its below
        it('the contract should be paused'); // Assert
        it('should emit the LogPaused event'); // Assert
    }
    
    context('if the contract is paused') {
        beforeEach(); // Arrange to set the test to  be paused
        it('should fail'); // Act and Assert
    }
}
```

Reply:
do you think I should add a test case at the same level like? thanks
```
context('When owner pause the contract') {
    context('if the contract is not paused') {}
    context('if the contract is paused') {}
    context('if the contract is unpaused') {} // Add new test case here?
}
```

Reply:
It think it is ok how you do it. I had difficulty following the wording of the `context`

Reply:
thanks @xavierlepretre I keep as it's for now.